### ISSUE TITLE
Add class for active pane

### ISSUE THREAD

**Issue by sudhakar** [ticlo/rc-dock#88]
Thank you for a wonderful library. BTW, I'm trying to style the tabs based on which panel is active. ie at any point in time, there could be atmost one pane active, so that active pane can be highlighted, something along the lines of 

<img width="830" alt="Screenshot 2021-05-13 at 10 32 17 PM" src="https://user-images.githubusercontent.com/6782/118160157-9329ea80-b43b-11eb-8d35-5aeb2fc888b2.png">

Active pane will change whenever focus is changed or z-index is changed either by clicking or via some api to activate the tab.

I tried using the controlled layout, but changes in z-index doesnt trigger onLayoutChange. Same with clicking inside a pane (& not changing active tab).

I tried adding `onMouseDownCapture` to the wrapper div of `DockLayout`, but `layout` for `dockbox` doesnt contain the bounds to determine which pane is activated. 

Is there any other way how this can be accomplished? If not, can you add a `active` class to the pane that has the focus?

**Comment by rinick** [comment]
have you tried the :focus-within selector?

you can use this to select the active panel, then use it as a parent selector to adjust tabs / content inside it

**Comment by sudhakar** [comment]
Beautiful! Didnt know `:focus-within` existed. Could solve my problem. Will try it out!

**Comment by sudhakar** [comment]
@rinick `:focus-within` works but requires `tabindex` to be set for the tabContent, tabBar etc. In addition to it, building a keyboard navigation is also difficult as `DockLayout.updateTab` activates the tab, but wouldnt set the focus.

IMHO, in addition to using `:focus-within`, if there's a additional class `panel-active` added to `dock-panel`, then it would be much easier to style. 

Here's [an example](https://wi-pane.stackblitz.io/) with [flexlayout-react](https://stackblitz.com/edit/wi-pane?file=app.tsx). You can use `LEFT`-`RIGHT` to move between the tabs & `UP`-`DOWN` to move between the panels. Would like to achieve the same effect with rc-dock. I like the cleaner api in rc-dock & would prefer to use it over the flexlayout.

Any help would be highly appreciated!

**Comment by rinick** [comment]
I might add arrow key navigation when focus is on tab button

but it wont be like your example.  actually that example is quite wrong, when mouse click in a panel, then move with arrow key,  I can see my cursor moving in the first panel, means focus is still in the first one, but colored border move to a different panel.  

this behavior might cause people to accidently press enter and submit a wrong form that he didn't expect
So I believe focus-within is the right way, 

maybe I can make panel to move focus on current tab when clicked

**Comment by sudhakar** [comment]
Agreed. It was mainly planned for read-only widgets in each of those tabs/panels. Without moving the focus, highlighting panels would create confusion.

Yes, moving focus to the tab when the tab is activated either by click or programatically would be great.
 
AFAIK, focus can only be given to elements with `tabindex`. What happens if user clicks on a text content which doesnt have `tabindex`? Will any `click` inside the tab would result in moving focus as well?

**Comment by sudhakar** [comment]
Here's an [example with rc-dock](https://stackblitz.com/edit/rc-dock?file=index.tsx) using `focus-within`. It does

1. Maximizes `Protect` tab when `alt+m` is pressed
2. Activate `Protect` tab when `alt+p` is pressed

It almost works except that,

1. Activating `Protect` tab doesn't give it the focus &
2. Even if `Protect` tab has the focus, `maximize/restore` makes it loose focus
3. It also needed `tabIndex={0}` on the first-child of `dock-tabpane` to capture focus when clicked inside text content
4. Clicking on tabbar still didn't give the focus to panel (as it doesnt have tabindex)

We can make the whole panel receive focus when anywhere inside the panel is clicked including the tabbar. This inturn will require tabpanel to receive focus, which IMO is the right thing to do for this (we can give `tabindex="-1` to `tab-panel`, so that usual tabbing wont give it focus, but can be focussed programatically)

> maybe I can make panel to move focus on current tab when clicked

So yes, making the panel receive focus when tab is clicked/activated will work

**Comment by rinick** [comment]
could you check 3.0.14

not sure if this update 100% solves this uses case, let me know if there is anything that you can't find a work around with current api

**Comment by rinick** [comment]
also Docklayout now has a new API
navigateToPanel()
if you call it with no parameter, it will just find the first active tab and focus on it

it could be useful to initialize the focus at the beginning, for example

```javascript
documentBodyKeyDownListener(e){
   if (e.key.startsWith('Arrow') && document.activeElement === document.body) {
     docklayout.navigateToPanel();  // move focus to any tab when the focus is on document body
   }
}
}
```

**Comment by sudhakar** [comment]
@rinick Tested with latest commit (https://github.com/ticlo/rc-dock/commit/3c51eb64d2ddf9bec00a84d1ddaa78cfce4ad3f6) & this is what I found so far

1. Tab navigation with `Arrow` keys moves focus sequentially between tabs, but incase of focus being in `floatbox` tab, it moves outside to dockbox as expected, but then once it goes into dockbox, it doest reverse back to `floatbox`
2. When the panel has active focus & if the panel is maximised, it looses focus. Same with restore as well
3. When the panel is maximised and if using `arrow` keys to navigate, focus moves outside maximised panel. Once it moves outside, it doesnt seem to loop back to maximised panel unless explicitly focus via click

**Comment by sudhakar** [comment]
Ahh ok. For 1, I see that you're finding the nearest tab adjacent to the active tab in the direction of the `Arrow`. So using the `Down` or `Up` key does take the focus into the `floatbox`. Beautiful!

**Comment by sudhakar** [comment]
For using `navigateToPanel(fromElement?: HTMLElement, direction?: string)`, one needs to store `ref` to the `HTMLElement` in their component. Instead, if it also takes `tabID` in-addition to `HTMLElement`, it would align with `updateTab` api signature. IMHO, for public api, having something like `focusPanel(tabId: string)` will be more nicer & `navigateToPanel` can be more of an internal API.

**Comment by rinick** [comment]
2)3) should be resolved, in 3.0.15 

but I don't quite get 4) and 5), why is keyboard moving so important on features that basically won't work without mouse?
I can't imagine that someone drags a panel, then wants to use arrow key to move to next tab

**Comment by rinick** [comment]
> For using `navigateToPanel(fromElement?: HTMLElement, direction?: string)`, one needs to store `ref` to the `HTMLElement` in their component. Instead, if it also takes `tabID` in-addition to `HTMLElement`, it would align with `updateTab` api signature. IMHO, for public api, having something like `focusPanel(tabId: string)` will be more nicer & `navigateToPanel` can be more of an internal API.

it can be used as public api mainly for the no parameter part.

but still, you can do this yourself if you want to focus on a panel
```javascript
document.querySelector(`.dock-panel[data-dockid="${this.panelToFocus}"] .dock-bar`)?.focus();
```

**Comment by sudhakar** [comment]
> but still, you can do this yourself if you want to focus on a panel
> ```document.querySelector(`.dock-panel[data-dockid="${this.panelId}"] .dock-bar`)?.focus();```

Agreed. 4 & 5 are not problems per se. It very well be done in the user-land either by doing `element.focus()` or overriding `z-index` using `:focus-within`

> I can't imagine that someone drags a panel, then wants to use arrow key to move to next tab

Usecase here is if someone uses `updateTab` or `dockMove` to activate or move panels. Even if someone drags a panel, it would be good for the tab or panel that was dragged to have the focus as they received a `TouchEvent` by the virtue of dragging.

> 2)3) should be resolved, in 3.0.15

2 & 3 will cover almost all the cases I had. BTW, thank you for awesome work on rc-tab :)
